### PR TITLE
User modal for signup and login

### PR DESCRIPTION
**Pull request by AlbertStjarne**
## PT:
https://www.pivotaltracker.com/story/show/161785282

## Includes:
* Add modal for signup and login on the landing page
* Add cancel functionality when clicking outside of the signup/login modal

## What we learnt:
* How to create a modal
* Cancel functionality when clicking outside of modal

image
image

### CODE REVIEW

File: app/controllers/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  def create
+    super
+  #   self.resource = warden.authenticate(auth_options)
+  #   if resource
+  #     sign_in(resource_name, resource)
+  #     flash[:notice] = 'Signed in successfully.'
+  #     redirect_to root_path
+  #   else
+  #     render json: { message: 'Could not log you in' }, status: :unauthorized
+  #   end
```
Comment (Rinbo) [inline app/controllers/registrations_controller.rb]: Seems like this can be removed

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  def create
+    super
+  #   self.resource = warden.authenticate(auth_options)
+  #   if resource
+  #     sign_in(resource_name, resource)
+  #     flash[:notice] = 'Signed in successfully.'
+  #     redirect_to root_path
+  #   else
+  #     render json: { message: 'Could not log you in' }, status: :unauthorized
+  #   end
```
Comment (Rinbo) [inline app/controllers/sessions_controller.rb]: Any need to keep this?

File: app/views/devise/registrations/new.js.erb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    signupForm = "<%= escape_javascript(render partial: 'partials/signup') %>"
+    modalLogin = document.getElementById('modal')
+} else {
+    let signupForm = "<%= escape_javascript(render partial: 'partials/signup') %>"
+    let displayElementLogin = document.getElementById('user-forms')
+    let modalLogin = document.getElementById('modal')
+}
+displayElementLogin.innerHTML = signupForm;
+
+window.onclick = function(event) {
```
Comment (Rinbo) [inline app/views/devise/registrations/new.js.erb]: this function is in booth reg/new.js and sessions/new.js. Can it be moved to application.js?

File: app/views/devise/sessions/new.js.erb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  loginForm = "<%= escape_javascript(render partial: 'partials/login') %>"
+  modalSignup = document.getElementById('modal')
+} else {
+  let loginForm = "<%= escape_javascript(render partial: 'partials/login') %>"
+  let displayElementSignup = document.getElementById('user-forms')
+  let modalSignup = document.getElementById('modal')
+}
+displayElementSignup.innerHTML = loginForm;
+
+window.onclick = function(event) {
```
Comment (Rinbo) [inline app/views/devise/sessions/new.js.erb]: See comment on reg/new.js

File: app/views/partials/_landing_nav.html.erb [hunk 5]
```diff
@@ -1,5 +1,9 @@
-<nav class="landing-nav">
+<%# <nav class="landing-nav">
   <form action="/users/sign_in" method="get">
     <button class="btn btn-white masthead-text login">Login</button>
   </form>
+</nav> %>
```
Comment (Rinbo) [inline app/views/partials/_landing_nav.html.erb]: Do we want this commented out code here?

File: app/views/partials/_landing_page.html.erb [hunk 6]
```diff
@@ -1,9 +1,12 @@
 <%= render 'partials/landing_nav' %>
 <section class="masthead">
   <h1 class="masthead-text heading">Sqedule</h1>
+  <div id="user-forms"></div>
   <div class="masthead-text sub-heading">Optimized staff and shift planning for small and medium sized businesses</div>
-  <form action="/users/sign_up" method="get">
-    <button class="btn btn-white masthead-text sign-up">Sign up</button>
-  </form>
+  <%# <form action="/users/sign_up" method="get"> %>
+    <%# <button class="btn btn-white masthead-text sign-up">Sign up</button> %>
```
Comment (Rinbo) [inline app/views/partials/_landing_page.html.erb]: More commented out code.

File: app/views/partials/_landing_page.html.erb [hunk 7]
```diff
@@ -1,9 +1,12 @@
...
+   <%= link_to 'Sign up', new_user_registration_path, remote: true, class: "sign-up btn btn-white masthead-text" %>
+
+  <%# </form> %>
```
Comment (Rinbo) [inline app/views/partials/_landing_page.html.erb]: Don't think we are gonna use this anymore?

File: app/controllers/registrations_controller.rb (hunk 1 above)
Comment (AlbertStjarne) [reply app/controllers/registrations_controller.rb]: correct

File: app/controllers/sessions_controller.rb (hunk 2 above)
Comment (AlbertStjarne) [reply app/controllers/sessions_controller.rb]: now removed

File: app/views/devise/registrations/new.js.erb (hunk 3 above)
Comment (AlbertStjarne) [reply app/views/devise/registrations/new.js.erb]: it can, it worked fine

File: app/views/partials/_landing_nav.html.erb (hunk 5 above)
Comment (AlbertStjarne) [reply app/views/partials/_landing_nav.html.erb]: removed

**Pull request merged**